tests/cephadm/test_ceph_health_detail_when_mon_offline.py

```diff
@@ -0,0 +1,72 @@
+from cli.cephadm.cephadm import CephAdm
+from cli.utilities.utils import bring_node_offline, is_node_online
+from utility.log import Log
+
+log = Log(__name__)
+
+
+class ClusterNotHealthy(Exception):
+    pass
+
+
+class NodeNotOffline(Exception):
+    pass
+
+
+class NodeNotOnline(Exception):
+    pass
+
+
+class UnexpectedHealthReport(Exception):
+    pass
+
+
+def run(ceph_cluster, **kw):
+    """Verify cluster health detail result when a node is down
+    Args:
+        **kw: Key/value pairs of configuration information to be used in the test.
+    """
+    installer_node = ceph_cluster.get_nodes(role="installer")[0]
+    mon_node = ceph_cluster.get_nodes(role="mon")[1]
+
+    # Get Ceph Health Detail
+    ceph_health = CephAdm(installer_node).ceph.health()
+    if not ceph_health == "HEALTH_OK":
+        raise ClusterNotHealthy("Ceph cluster not healthy")
+
+    # Bring the mon_node down
+    out = bring_node_offline(mon_node, installer_node, timeout=30)
+    if not out:
+        raise NodeNotOffline(f"Failed to bring {mon_node.hostname} offline")
+
+    # Once the mon node is offline, check for the cluster health detail
+    ceph_health_detail = CephAdm(installer_node).ceph.health(detail=True)
```

Comment:
Will logs get generated immediately ? No need to wait here ? 

As here check is for absence of logs, better to wait till node get up and get logs for time period of node was down.

Reply:
As per the bz, the health detail command has to be executed while the node is down. In that case, will introduce a sleep less than the network interface down time and verify the result

Reply:
Added a sleep of 10.

Reply:
Now in this case need to get logs from last 10 secs